pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -261,7 +264,33 @@ func NewContainerManager(mountUtil mount.Interface, cadvisorInterface cadvisor.I
 		cgroupRoot:          cgroupRoot,
 		recorder:            recorder,
 		qosContainerManager: qosContainerManager,
-	}, nil
+	}
+
+	updateDeviceCapacityFunc := func(updates v1.ResourceList) {
+		cm.Lock()
+		defer cm.Unlock()
+		for k, v := range updates {
+			if v.Value() <= 0 {
+				delete(cm.capacity, k)
+			} else {
+				cm.capacity[k] = v
+			}
+		}
+	}
+
+	glog.Infof("Creating device plugin handler: %t", devicePluginEnabled)
+	var devicePluginHandler DevicePluginHandler
+	if devicePluginEnabled {
+		devicePluginHandler, err = NewDevicePluginHandlerImpl(updateDeviceCapacityFunc)
+	} else {
+		devicePluginHandler, err = NewDevicePluginHandlerStub()
+	}
+	if err != nil {
+		return nil, err
+	}
+	cm.devicePluginHandler = devicePluginHandler
```

Comment:
Ideally, there will be a `NewDevicePluginHandler()` method that will be invoked from `cmd/kubelet` package and a devicepluginhandler object will be injected into container manager. I'm ok with this model for now, but in the long run, I'd prefer not creating objects from within the kubelet library.